### ISSUE TITLE
TOTP QR code scanner causes crash

### ISSUE THREAD

**Issue by lschober-bw** [bitwarden/mobile#975]
## Describe the Bug

If you attempt to use the QR code scanner for TOTP, and Bitwarden is not already given Camera permissions, the system will ask you to allow the application access to the camera. This action causes the system and application to bind up and become unresponsive.

## Steps To Reproduce

1. If enabled, go to the system settings and revoke Bitwarden's camera permission. 
2. Open Bitwarden
3. Edit any Login type item
4. Select the camera icon in the Authenticator Key (TOTP) field
5. Grant Bitwarden camera access.

## Expected Result

The camera is enabled and you can scan your QR code.

## Actual Result

The camera viewfinder is stuck as a white screen. The application and system become unresponsive to input. The application crashes.

## Environment

- Device: Pixel 2XL
- Operating system: Android 10 (QQ2A.200501.001.B3)
- Build Version: 2.4.3 (2523)
- Is this a Beta release? Y

## Additional Context

Once the application restarts with the camera permission enabled, scanning a QR code is successful.

**Comment by bcronce** [comment]
I'm stuck on Android 8 and was in darkmode, when instead of a white screen, I had a black one. I assume this is because dark mode.

Also mention that I just kept trying and eventually got it to take. Seemed like every other time crashed then worked. But it's late and I don't want to setup any more tonight to see if that pattern persists.

**Comment by cscharf** [comment]
Resolved by #1005

**Comment by OLLI-S** [comment]
@cscharf 
I switched to Bitwarden some days ago and still had this issue the last days (the Android app crashed while scanning the QR code).
So this issue is not fixed!

**Comment by julian-alarcon** [comment]
I have the same error mentioned by @OLLI-S . But I think that this is a different issue than the reported bug here as the App already have the Camera permission and when I try to set the code scanning the QR the App crashes.

**Comment by gaticssa** [comment]
I have (had?) the same problem on a Galaxy S10e. 

The first time I tried to scan a QR code (for an Amazon account) the Bitwarden app asked for the Camera permission.
I allowed that, but still the app crashed on every scanning attempt.

Then I noticed under App Settings that the Bitwarden app did not have the Storage permission. I don't recall whether it had asked for that on install, but it definitely didn't ask when scanning the QR code.

I allowed that, and I was able to successfully scan at least that one code. I haven't tried with any other sites, so I'm not sure if it was just a lucky coincidence that it worked.

Bitwarden app: Version 2.6.1 (3178)
Android Build Number: QP1A.190711.020.G970FXXU9DTJA
